packages/block-library/src/template-part/edit/selection-modal/patterns-list.js

```diff
@@ -1,64 +1,73 @@
 /**
  * WordPress dependencies
  */
-import { __experimentalBlockPatternSetup as BlockPatternSetup } from '@wordpress/block-editor';
-import { useEffect, useState } from '@wordpress/element';
+import {
+	__experimentalBlockPatternsList as BlockPatternsList,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { useState, useMemo } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 import {
 	TextControl,
 	Flex,
 	FlexItem,
 	Button,
 	Modal,
-	Placeholder,
 } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useAsyncList } from '@wordpress/compose';
 
-export default function PatternsSetup( {
+export default function PatternsList( {
 	area,
 	areaLabel,
-	areaIcon,
 	clientId,
-	onCreate,
-	resetPlaceholder,
+	onSelect,
 } ) {
 	const blockNameWithArea = area
 		? `core/template-part/${ area }`
 		: 'core/template-part';
+	const blockPatterns = useSelect(
```

Comment:
The list of the patterns was created inside  BlockPatternSetup. Do you think we still have the possibility of a reusable component for selecting patterns like BlockPatternSetup?

Reply:
I'm not sure, I left the current one because it's still used for other blocks (query...) but once we move to that block, we'd need to see.

For now I lean towards that not being needed but we'll see. The reason is that the template part modal is a bit different in the sense that it's composed of two areas (patterns and template parts) and the behavior of creating is a bit different between the two.

Reply:
> For now I lean towards that not being needed but we'll see. The reason is that the template part modal is a bit different in the sense that it's composed of two areas (patterns and template parts) and the behavior of creating is a bit different between the two.

Maybe a solution would be for the template part to have two buttons choose a template part, and choose a pattern? But that makes the replace strange as we don't want two replace buttons.

Reply:
Yeah, I initially suggested that but @jameskoster thought it was simpler to just merge everything.

Reply:
Yup, I think the quicker we can show the user some options, the better.

Reply:
>Maybe a solution would be for the template part to have two buttons choose a template part, and choose a pattern?

That is the current way, no? Also I agree that Template Part flow is different from other blocks which most commonly are uncontrolled blocks.